pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -100,54 +86,196 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
+	// get permission type
+	permission := &managedv1alpha1.Permission{}
+
+	// get clusterRoleName from permission type
+	clusterRoleName := permission.ClusterRoleName
+
+	// get groupname from GroupPermission object
+	groupName := instance.Spec.GroupName
+
+	var tmpArray []string
+
+	for _, a := range instance.Spec.ClusterPermissions {
+		clusterRoleName := a.ClusterRoleName
+		clusterRoleBindingName := clusterRoleName + "-" + groupName
+
+		tmpArray = append(tmpArray, clusterRoleBindingName)
+	}
+
+	// clusterRoleBindingItems := clusterRoleBindingList.Items
+
+	if checkClusterRoleBindingName(tmpArray, clusterRoleBindingList) {
+		return reconcile.Result{}, nil
+	} else {
+		// create a clusterRoleBinding
+		// this is absolutely NOT gonna work... plz advice... TODO
+		newCRB := newClusterRoleBinding(instance, permission)
+		if newCRB != nil {
+			groupPermissionConditions := instance.Status.Conditions
+
+			// make new condition
+			newCondition := managedv1alpha1.Condition{
+				LastTransitionTime: metav1.Now(),
+				ClusterRoleName:    clusterRoleName,
+				Status:             true,
+				State:              "Created",
+			}
+
+			// append new condition to conditions array
+			groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+			// update GroupPermission object with new conditions
+			instance.Status.Conditions = groupPermissionConditions
+
+			err = r.client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				reqLogger.Error(err, "Failed to update condition.")
+				return reconcile.Result{}, nil
+			}
+		} else {
+			groupPermissionConditions := instance.Status.Conditions
+
+			// make new condition
+			newCondition := managedv1alpha1.Condition{
+				LastTransitionTime: metav1.Now(),
+				Message:            "Unable to create ClusterRoleBinding: <error from platform>",
+				ClusterRoleName:    clusterRoleName,
+				Status:             true,
+				State:              "Failed",
+			}
+
+			// append new condition to conditions array
+			groupPermissionConditions = append(groupPermissionConditions, newCondition)
+
+			// update GroupPermission object with new conditions
+			instance.Status.Conditions = groupPermissionConditions
+
+			err = r.client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				reqLogger.Error(err, "Failed to update condition.")
+				return reconcile.Result{}, nil
+			}
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
 
-// newPodForCR returns a busybox pod with the same name/namespace as the cr
-func newPodForCR(cr *managedv1alpha1.GroupPermission) *corev1.Pod {
-	labels := map[string]string{
-		"app": cr.Name,
-	}
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
+// newClusterRoleBinding creates and returns ClusterRoleBinding
+func newClusterRoleBinding(groupPermission *managedv1alpha1.GroupPermission, permission *managedv1alpha1.Permission) *v1.ClusterRoleBinding {
```

Comment:
Move the *managedv1alpha1.GroupPermission to a string since you only use the name

Both the Subject and and RoleFed must have a kind filled in